packages/client/hmi-client/src/components/mathml/tera-math-editor.vue

```diff
@@ -0,0 +1,262 @@
+			<div style="align-self: center">[Math Renderer]</div>
+			<div class="math-options">
+				<label>
+					<input type="radio" v-model="mathMode" :value="MathEditorModes.JAX" />
+					MathJAX
+				</label>
+				<label>
+					<input type="radio" v-model="mathMode" :value="MathEditorModes.LIVE" />
+					MathLIVE
+				</label>
+			</div>
+		</section>
+		<section v-if="mathMode === MathEditorModes.LIVE">
+			<math-field
+				class="mathlive-equation"
+				ref="mathLiveField"
+				virtual-keyboard-mode="false"
+				:disabled="!isEditingEq"
+				:style="{ borderColor: isMathMlValid ? 'inherit' : 'red' }"
+				><slot v-if="mathMode === MathEditorModes.LIVE"></slot
+			></math-field>
+		</section>
+		<section class="mathjax-container" v-else-if="mathMode === MathEditorModes.JAX">
+			<Textarea
+				v-model="jaxEquation"
+				class="mathjax-equation"
+				id="mathjax"
+				type="text"
+				rows="2"
+				aria-label="mathjax"
+				:disabled="!isEditingEq"
+				autoResize
+			/>
+			<div><vue-mathjax :formula="jaxEquationString" /></div>
+		</section>
+	</section>
+</template>
+
+<script setup lang="ts">
+import { ref, watch, onUpdated, computed, onMounted } from 'vue';
+import { Mathfield } from 'mathlive';
+import Mathml2latex from 'mathml-to-latex';
+import { logger } from '@/utils/logger';
+import Textarea from 'primevue/textarea';
+import { isMathML, MathEditorModes } from '@/utils/math';
+import Button from 'primevue/button';
+
+const mathLiveField = ref<Mathfield | null>(null);
+const latexEquation = ref<string | undefined>('');
+const jaxEquation = ref<string | undefined>('');
+const mathMode = ref<string | null>(null);
+
+const jaxEquationString = computed((): string =>
+	latexEquation.value ? `$$${latexEquation.value}$$` : ''
+);
+
+const emit = defineEmits(['equation-updated', 'validate-mathml', 'cancel-editing']);
+
+const props = defineProps({
+	// LaTeX formula to be populated
+	latexEquation: {
+		type: String,
+		required: true
+	},
+	// Math Renderer mode
+	mathMode: {
+		type: String as () => MathEditorModes,
+		default: MathEditorModes.LIVE
+	},
+	// Show edit button
+	isEditable: {
+		type: Boolean,
+		default: true
+	},
+	// Show edit button
+	isEditingEq: {
+		type: Boolean,
+		default: true
+	},
+	// Show the renderer selection box
+	showDevOptions: {
+		type: Boolean,
+		default: false
+	},
+	// check if the mathml is valid
+	isMathMlValid: {
+		type: Boolean,
+		default: true
+	}
+});
+
+const getLaTeX = (equationString: string): string => {
+	if (isMathML(equationString)) {
+		try {
+			return Mathml2latex.convert(equationString);
+		} catch (error) {
+			logger.error(error, { showToast: false, silent: true });
+			return '';
+		}
+	}
+
+	return equationString;
+};
+
+onMounted(() => {
+	if (props.mathMode) {
+		mathMode.value = props.mathMode;
+	}
+});
+
+onUpdated(() => {
+	if (mathMode.value === MathEditorModes.LIVE) {
+		mathLiveField.value?.setValue(latexEquation.value);
+	}
+});
```

Comment:
Is this life-hook cycle necessary? Seems like you are doing the equivalent checks in the watchers already?

Reply:
It actually is.  I spent way too time trying to get this to work without doing this 😭